#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.4851920955529382 0.6110588898515221 0.6254563642224562 1.807194496648216
  position 0.15100855432549362 -0.3029580233717593 1.9748593102689291
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 1 1 0
  floorSize 2 2
}
Robot {
  translation 0.1 0.1 0
  rotation 0 0 1 3.14159
  children [
    GPS {
    }
    DEF body Transform {
      translation 0 0 0.0415
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 14.809636955439288
        anchor 0.045 0 0.025
      }
      device [
        RotationalMotor {
          name "motor_1"
        }
        PositionSensor {
          name "ps_1"
        }
      ]
      endPoint Solid {
        translation 0.045 0 0.025
        rotation 0.6695916733809867 0.32250191418081736 0.6690586717833936 2.5171856395186225
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        boundingObject USE wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 37.23808496491429
        anchor -0.045 0 0.025
      }
      device [
        RotationalMotor {
          name "motor_2"
        }
        PositionSensor {
          name "ps_2"
        }
      ]
      endPoint Solid {
        translation -0.045 0 0.025
        rotation -0.22290632583766148 0.949054588194654 -0.22272889019664613 1.6222662198297535
        children [
          USE wheel
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics USE wheel_physics
      }
    }
    Solid {
      translation 0.015 -0.045 0.07
      children [
        DEF eye Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      name "solid(2)"
    }
    Solid {
      translation -0.015 -0.045 0.07
      children [
        USE eye
      ]
      name "solid(3)"
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "gps-controller"
}
